core/src/test/java/ai/timefold/solver/core/impl/move/streams/maybeapi/provider/ChangeMoveProviderTest.java

```diff
@@ -84,6 +95,133 @@ void fromSolutionBasicVariable() {
         });
     }
 
+    @Test
+    void fromSolutionBasicVariableIncompleteValueRange() {
```

Comment:
This test addresses the question I had regarding `ChangeMoveProvider`. I am not entirely convinced that we should apply the filter in any circumstances. So, here are two questions:
1. Can we identify unused problem fact generators and exclude them from the related streams?
2. Can we fail fast if any unused fact generators like `getValueListNotInValueRange` are identified?
My main concern is to avoid filtering values in large datasets, even with the optimizations from `ListValueRange` being merged.